### PR TITLE
tests: shell-test: use count for REP

### PR DESCRIPTION
**Pull request by blueyed**

### CODE REVIEW

File: test/functional/terminal/edit_spec.lua [hunk 1]
```diff
@@ -34,19 +34,18 @@ describe(':edit term://*', function()
   it("runs TermOpen early enough to set buffer-local 'scrollback'", function()
     local columns, lines = 20, 4
     local scr = get_screen(columns, lines)
-    local rep = 'a'
+    local rep = 97
     meths.set_option('shellcmdflag', 'REP ' .. rep)
     command('set shellxquote=')  -- win: avoid extra quotes
-    local rep_size = rep:byte()  -- 'a' => 97
```
Comment (justinmk) [inline test/functional/terminal/edit_spec.lua]: this was intentional. why remove it?

File: test/functional/fixtures/shell-test.c [hunk 2]
```diff
@@ -71,8 +70,12 @@ int main(int argc, char **argv)
         fprintf(stderr, "Not enough REP arguments\n");
         return 4;
       }
-      uint8_t number = (uint8_t) *argv[2];
-      for (uint8_t i = 0; i < number; i++) {
+      int count = 0;
+      if (sscanf(argv[2], "%d", &count) != 1) {
```
Comment (justinmk) [inline test/functional/fixtures/shell-test.c]: while you're in the area, PVS had this complaint about the other scanf() in this file :)

```
./test/functional/fixtures/shell-test.c:107:1: warning: V576 Incorrect format. Consider checking the third actual argument of the 'sscanf' function. It's dangerous to use string specifier without width specification. Buffer overflow is possible.
```

Comment (blueyed) [reply test/functional/fixtures/shell-test.c]: Done.

**Review by justinmk — APPROVED** [review]

File: test/functional/fixtures/shell-test.c (hunk 2 above)
Comment (blueyed) [reply test/functional/fixtures/shell-test.c]: JFI: `cppcheck` also warns about it:

> [test/functional/fixtures/shell-test.c:107]: (warning) sscanf() without field width limits can crash with huge input data.

**Pull request merged**